docs/articles/documentation/recipes/best-practices/create-faster-tests.md

```diff
@@ -0,0 +1,101 @@
+---
+layout: docs
+title: Create Faster Tests
+permalink: /documentation/recipes/best-practices/create-faster-tests.html
+redirect_from:
+  - /documentation/recipes/create-faster-tests.html
+---
+# Speed Up Your Tests
+
+The speed of TestCafe tests depends on the loading speed of your web application. The best way to speed up your tests may be to optimize the tested application. Make sure that your application doesn't load large payloads of unused CSS or scripts.
+
+Test in an environment with performance headroom. Lack of resources may increase the time browsers take to initialize, connect to TestCafe and load the tested application. If you have to test in a resource-low environment, [run tests in headless browsers](#run-tests-in-headless-browsers).
+
+For better performance, launch tests in local browsers. Network lag between TestCafe and remote browsers causes delays in test execution.
+
+Remote browsers may take a long time to initialize and connect to TestCafe. Use the  [--browser-init-timeout](../../reference/command-line-interface.md#--browser-init-timeout-ms) CLI option to set a time limit for browsers to connect. Tests fail if this timeout is exceeded - it might be a good time to debug your remote browser connection.
+
+TestCafe's [speed](../../reference/command-line-interface.md#--speed-factor) option allows you to change test speed. The default value is `1`, which is fastest.
```

Comment:
I don't have any insight into how the "speed" option actually implemented, but from my point of view the "speed" option specified in the CLI, API, configuration file or set using the [setTestSpeed](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/settestspeed.html) method overrides the corresponding option of the emulation action (the speed with which this action occur) and affects the speed of the emulated cursor movement (the delay between the emulation actions, in fact).

I would approach this part in a different way:
> TestCafe can emulate a "human-like" interaction with a tested page by imitating delays with which certain actions are emulated on the page. It can be achieved by specifying the "speed" option on the test run level (CLI, API, config file), the test level (t.setTestSpeed) or level of the separate action. But if your concern is speed of test execution, you probably don't want waste time on it. So you should check that the "speed" option is set to the maximum possible value - `1`, which is set by default.

Reply:
> TestCafe emulates real user actions on tested webpages. [Speed](../../reference/command-line-interface.md#--speed-factor) option enables you to change action emulation speed.

> The highest value of `1` represents the fastest actions on the page. This is a default value. The minimum value of `0.01` represents a speed that is 100 lower. Setting a lower `speed` can be useful for debugging tests, but slows tests down.

> If your tests have `speed` set in the run configuration, disable this setting or set it to `1`.